### PR TITLE
User facing errors and tests for package operations

### PR DESCRIPTION
**Pull request by Naatan**
<table>
<td>
  <a href="https://activestatef.atlassian.net/browse/DX-2149" title="DX-2149" target="_blank">StoryDX-2149</a>  User facing errors for `state install`
  </td></table>
  <br />

### CODE REVIEW

File: internal/runbits/requirements/rationalize.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	case errors.As(*err, &noMatchesErr) && noMatchesErr.Alternatives != nil:
+		*err = errs.WrapUserFacing(*err,
+			locale.Tr("package_ingredient_alternatives", tooManyMatchesErr.Query, *noMatchesErr.Alternatives),
+			errs.SetInput())
+
+	// We communicate buildplanner errors verbatim as the intend is that these are curated by the buildplanner
+	case errors.As(*err, &buildPlannerErr):
+		*err = errs.WrapUserFacing(*err,
+			buildPlannerErr.LocalizedError(),
+			errs.SetIf(buildPlannerErr.InputError(), errs.SetInput()))
```
Comment (mitchell-as) [inline internal/runbits/requirements/rationalize.go:51]: Wow, metaprogramming for error handling!

File: test/integration/install_int_test.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
+package integration
```
Comment (mitchell-as) [inline test/integration/install_int_test.go:1]: There's a nice set of tests here :)

**Review by mitchell-as — CHANGES_REQUESTED** [review]
Looks great! A couple of minor things to think about.

File: internal/runbits/requirements/rationalize.go (hunk 1 above)
Comment (Naatan) [reply internal/runbits/requirements/rationalize.go:51]: I'll be a functional programmer yet..

**Review by mitchell-as — DISMISSED** [review]

**Pull request merged**